File: PSME/lui/OS/rootfs/opt/intel/rackscale/include/psme_xml_structure/model/memory.py
"""
 * @section LICENSE
 *
 * @copyright
 * Copyright (c) 2015-2018 Intel Corporation
 *
 * @copyright
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * @copyright
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * @copyright
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * @section DESCRIPTION
"""
from include.psme_xml_structure.model.psme_structure_element import PsmeStructureElement

from include.psme_xml_structure.model.fru_info import FruInfo
from include.psme_xml_structure.model.status import Status
from include.psme_xml_structure.model.dimmLocation import DimmLocation


class Memory(PsmeStructureElement):
    def __init__(self):
        self.dimmLocation = DimmLocation()
        self.deviceLocator = ""
        self.rankCount = ""
        self.operatingSpeedMHz = 0
        self.minimumVoltageVolt = 0
        self.maximumVoltageVolt = 0
        self.voltageVolt = 0
        self.dataWidthBits = 0
        self.busWidthBits = 0
        self.fruInfo = FruInfo()

File: PODM/SW/node-composer/src/main/java/com/intel/rsd/nodecomposer/persistence/redfish/Memory.java
/*
 * Copyright (c) 2016-2019 Intel Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.intel.rsd.nodecomposer.persistence.redfish;

import com.intel.rsd.nodecomposer.persistence.redfish.base.DiscoverableEntity;
import com.intel.rsd.nodecomposer.persistence.redfish.base.MemoryModule;
import com.intel.rsd.nodecomposer.types.MemoryDeviceType;
import com.intel.rsd.nodecomposer.types.MemoryType;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Enumerated;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import java.util.Objects;

import static javax.persistence.CascadeType.MERGE;
import static javax.persistence.CascadeType.PERSIST;
import static javax.persistence.EnumType.STRING;
import static javax.persistence.FetchType.LAZY;

@javax.persistence.Entity
@Table(name = "memory")
public class Memory extends DiscoverableEntity implements MemoryModule {
    @Getter
    @Setter
    @Column(name = "memory_device_type")
    @Enumerated(STRING)
    private MemoryDeviceType memoryDeviceType;

    @Getter
    @Setter
    @Column(name = "memory_type")
    @Enumerated(STRING)
    private MemoryType memoryType;

    @Getter
    @Setter
    @Column(name = "capacity_mib")
    private Integer capacityMib;

    @Getter
    @Setter
    @Column(name = "data_width_bits")
    private Integer dataWidthBits;

    @Getter
    @Setter
    @Column(name = "manufacturer")
    private String manufacturer;

    @Getter
    @Setter
    @Column(name = "operating_speed_mhz")
    private Integer operatingSpeedMhz;

    @ManyToOne(fetch = LAZY, cascade = {MERGE, PERSIST})
    @JoinColumn(name = "computer_system_id")
    private ComputerSystem computerSystem;

    public ComputerSystem getComputerSystem() {
        return computerSystem;
    }

    public void setComputerSystem(ComputerSystem computerSystem) {
        if (!Objects.equals(this.computerSystem, computerSystem)) {
            unlinkComputerSystem(this.computerSystem);
            this.computerSystem = computerSystem;
            if (computerSystem != null && !computerSystem.getMemoryModules().contains(this)) {
                computerSystem.addMemoryModule(this);
            }
        }
    }

    public void unlinkComputerSystem(ComputerSystem computerSystem) {
        if (Objects.equals(this.computerSystem, computerSystem)) {
            this.computerSystem = null;
            if (computerSystem != null) {
                computerSystem.unlinkMemoryModule(this);
            }
        }
    }

    @Override
    public void preRemove() {
        unlinkComputerSystem(computerSystem);
    }
}
